src/spdx/parser/rdf/snippet_parser.py

```diff
@@ -0,0 +1,78 @@
+    file_spdx_id_uri = graph.value(subject=snippet_node, predicate=SPDX_NAMESPACE.snippetFromFile)
+    file_spdx_id = parse_spdx_id(file_spdx_id_uri, doc_namespace, graph)
+    byte_range = parse_ranges(snippet_node, graph, POINTER_NAMESPACE.ByteOffsetPointer, POINTER_NAMESPACE.offset)
+    line_range = parse_ranges(snippet_node, graph, POINTER_NAMESPACE.LineCharPointer, POINTER_NAMESPACE.lineNumber)
+    license_concluded = parse_literal_or_no_assertion_or_none(
+        logger, graph, snippet_node, SPDX_NAMESPACE.licenseConcluded,
+        parsing_method=lambda x: parse_license_expression(x, graph, doc_namespace))
+    license_info_in_snippet = []
+    for (_, _, license_info_in_snippet_node) in graph.triples(
+        (snippet_node, SPDX_NAMESPACE.licenseInfoInSnippet, None)):
+        license_info_in_snippet.append(
+            get_correct_typed_value(logger, license_info_in_snippet_node,
+                                    lambda x: parse_license_expression(x, graph, doc_namespace)))
+    license_comment = parse_literal(logger, graph, snippet_node, SPDX_NAMESPACE.licenseComments)
+    copyright_text = parse_literal_or_no_assertion_or_none(logger, graph, snippet_node, SPDX_NAMESPACE.copyrightText,
+                                                           parsing_method=str)
+    comment = parse_literal(logger, graph, snippet_node, RDFS.comment)
+    name = parse_literal(logger, graph, snippet_node, SPDX_NAMESPACE.name)
+    attribution_texts = []
+    for (_, _, attribution_text_literal) in graph.triples((snippet_node, SPDX_NAMESPACE.attributionText, None)):
+        attribution_texts.append(attribution_text_literal.toPython())
+
+    raise_parsing_error_if_logger_has_messages(logger, "Snippet")
+    snippet = construct_or_raise_parsing_error(Snippet,
+                                               dict(spdx_id=spdx_id, file_spdx_id=file_spdx_id, byte_range=byte_range,
+                                                    line_range=line_range, license_concluded=license_concluded,
+                                                    license_info_in_snippet=license_info_in_snippet,
+                                                    license_comment=license_comment,
+                                                    copyright_text=copyright_text, comment=comment, name=name,
+                                                    attribution_texts=attribution_texts))
+    return snippet
+
+
+def parse_ranges(snippet_node: URIRef, graph: Graph, pointer: Node, member: Node) -> Tuple[int, int]:
+    start_range = None
+    end_range = None
+    for (_, _, start_end_pointer) in graph.triples((snippet_node, SPDX_NAMESPACE.range, None)):
+        for (_, _, pointer_node) in graph.triples((start_end_pointer, POINTER_NAMESPACE.startPointer, None)):
+            for (_, _, _) in graph.triples((pointer_node, RDF.type, pointer)):
+                start_range = parse_range_value(graph, pointer_node, member)
```

Comment:
This for-loop does not make sense as it does not use any internal variables.
Is this implemented in case there are multiple range values given? In that case it would probably be better to let the `UniquenessError` be raised and excepted.

Same goes for the analogous for loop for the endPointer.